### PR TITLE
chore: add unit test coverage and enforcement

### PR DESCRIPTION
**Pull request by ssvegaraju**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+coverage:
+  round: down
+  precision: 2
+  range: '90...100'
+  status:
+    project:
+      default:
+        # basic
+        target: 90%
```
Comment (carvantes) [inline codecov.yml]: It looks like the codecov setup is not working.

These same changes were done in the search package (https://github.com/awslabs/fhir-works-on-aws-search-es/pull/147) but there are no Codecov reports yet: https://app.codecov.io/gh/awslabs/fhir-works-on-aws-search-es

Also, Codecov is not posting coverage messages on PRs like https://github.com/awslabs/service-workbench-on-aws/pull/844#issuecomment-998888816

---

Looks like we cannot use  target of 90% for all our packages. Looking at https://github.com/awslabs/fhir-works-on-aws-routing/runs/4529235534?check_suite_focus=true, this package has  test coverage of 75%. Also, looks like the coverage target is not being enforced (I think this PR should fail the 90% goal). We may need to set up different coverage goals for each package to get started.

File: .github/workflows/code-coverage.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+#
+
+name: Code Coverage Report
+
+on: [push, pull_request]
+
+jobs:
+  test:
```
Comment (carvantes) [inline .github/workflows/code-coverage.yml]: consider a more meaningful job name. Maybe `Report test coverage`

**Review by carvantes — CHANGES_REQUESTED** [review]

File: .github/workflows/code-coverage.yml (hunk 2 above)
Comment (ssvegaraju) [reply .github/workflows/code-coverage.yml]: Updated the name!

File: codecov.yml (hunk 1 above)
Comment (ssvegaraju) [reply codecov.yml]: Looks like the issue was the coverageReporters defined in the package.json. Removing that and updating coverage seems to have worked.

**Review by carvantes — APPROVED** [review]

**Pull request merged**